OrbitCaptureClient/CaptureClient.cpp

```diff
@@ -71,14 +91,35 @@ void CaptureClient::Capture(
   FinishCapture();
 }
 
-void CaptureClient::StopCapture() {
+bool CaptureClient::StopCapture() {
+  absl::MutexLock lock(&state_mutex_);
+  if (state_ == State::kStarting) {
+    // Block and wait until the state is not kStarting
+    bool (*IsNotStarting)(State * state) = [](State* state) {
+      return *state != State::kStarting;
+    };
+    state_mutex_.Await(absl::Condition(IsNotStarting, &state_));
+  }
+
+  if (state_ != State::kStarted) {
+    return false;
+  }
+
   CHECK(reader_writer_ != nullptr);
 
   if (!reader_writer_->WritesDone()) {
-    ERROR("Finishing writing on Capture's gRPC stream");
-    FinishCapture();
+    // It looks like there is no way to recover from that
+    // we can't safely reset reade_writer_ here because it
+    // will likely lead to null pointer dereference on the
+    // capture thread on reader_writer_->Read(). And no good
+    // way to stop the capture thread as well, since we do not
+    // know if this is going to be blocked on the same call
+    // forever. This is the best we can do here:
+    FATAL("WritesDone failed unable to finish the capture in orderly manner");
```

Comment:
This will lead to Orbit crashes, right? Shouldn't it be handled by the UI in some way?

Reply:
well I do not see how we can end-up in consistent state. I tried to explain it in the comment above.

Reply:
Okay. Have you considered using Check then? However, either is fine for me.

Reply:
I think this error message is more expressive than Check failed, and CHECK is for something that shouldn't happen if it happens this is programming error.

In this case the situation is a bit different. This can happen, we just cannot recover from it.

Reply:
Okay good point (even though I though CHECK can also have a message, like assert would), so yes CHECK does not make sense. 
If this can occur, I would still at least inform the user that something bad happend and that Orbit should be restarted.

Reply:
I see your point. I will add todo for now, This will probably require another callback. OnAborted/Fatal to let client display a message before we abort.

Reply:
Okay cool. Thanks again!